include/mbedtls/entropy.h

```diff
@@ -95,7 +95,7 @@ extern "C" {
  *
  * \param data      Callback-specific data pointer
  * \param output    Data to fill
- * \param len       Maximum size to provide
+ * \param len       Maximum size to provide, must be at most MBEDTLS_ENTROPY_BLOCK_SIZE.
```

Comment:
I have several concerns here:

1. This is the entropy callback used by the entropy pool and this documentation shouldn't be changed. 

The callback we need to document is in `hmac_drbg.h` and `ctr_drbg.h` in the context and as a parameter to the corresponding seed function.

2. Also, since this is a documentation of a callback that people will implement, I think it would be best to
- Make it clear if we are imposing requirements on the caller or the implementer
- Present it from the implementers point of view, as a requirement on the implementation

I suggest something like:
> \warning \p f_entropy MUST fill the buffer with full entropy and MUST be able to do so for buffer lengths up to at least `MBEDTLS_ENTROPY_BLOCK_SIZE` bytes.

3. Even if we don't change our entropy function here and now I agree that it is the right thing moving forward. Getting there would be much smoother if we would require implementors to handle arbitrary requests (even if our implementation does not do it yet), therefore I suggest that we do so on development and keep the milder requirement on the backport branches.

Reply:
>1. This is the entropy callback used by the entropy pool and this documentation shouldn't be changed.

I disagree. Since this entropy function is used by our drbg functions, we expect implementors this callback accept `MBEDTLS_ENTROPY_BLOCK_SIZE`  at most.

>1. Also, since this is a documentation of a callback that people will implement, I think it would be best to
> * Make it clear if we are imposing requirements on the caller or the implementer
> * Present it from the implementers point of view, as a requirement on the implementation

This is why I think that the warning should be in the entropy callback description

Reply:
> I disagree. Since this entropy function is used by our drbg functions, we expect implementors this callback accept `MBEDTLS_ENTROPY_BLOCK_SIZE` at most.

No, this function is used by our entropy pool and not our drbg functions:
https://github.com/ARMmbed/mbedtls/blob/9c99dc862cd5cf9cdc62cbfcd200063c83295397/library/entropy.c#L143-L145

Here is the function used by our DRBG functions:
https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/ctr_drbg.h#L143
https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/ctr_drbg.h#L183
https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/hmac_drbg.h#L98
https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/hmac_drbg.h#L142

Reply:
Thanks for the correction, for some reason I confused the entropy source function callback with the entropy function callback. I'll fix that